rancher2/structure_catalog_v2.go

```diff
@@ -74,11 +74,12 @@ func expandCatalogV2(in *schema.ResourceData) *ClusterRepo {
 	if v, ok := in.Get("enabled").(bool); ok {
 		obj.Spec.Enabled = &v
 	}
-	if v, ok := in.Get("git_branch").(string); ok {
-		obj.Spec.GitBranch = v
-	}
-	if v, ok := in.Get("git_repo").(string); ok {
+	if v, ok := in.Get("git_repo").(string); ok && len(v) > 0 {
 		obj.Spec.GitRepo = v
+		obj.Spec.GitBranch = catalogV2DefaultGitBranch
+	}
+	if v, ok := in.Get("git_branch").(string); ok && len(v) > 0 {
```

Comment:
Should this move up, nested in `git_repo` check or is there a need to be able to specify `git_branch` outside of using `git_repo`? I know this is probably caught with the added conflicts but code-wise I can't see the logic why it is not nested.

Reply:
The problem here was with the default value assigned by tfp. As `git_branch` had a default value, it was adding it to all catalogs v2, `git_repo` and `url` types. The logic here is that `git_branch` got a default value, just if `git_repo` has been specified. If `git_branch` has been also specified in the tf file, then the defined value is overwriting the default value.